spring-cloud-netflix-core/src/main/java/org/springframework/cloud/netflix/zuul/filters/ProxyRequestHelper.java

```diff
@@ -129,8 +131,14 @@ public void setResponse(int status, InputStream entity,
 			RequestContext.getCurrentContext().setResponseDataStream(entity);
 		}
 		boolean isOriginResponseGzipped = false;
-		if (headers.containsKey(CONTENT_ENCODING)) {
-			Collection<String> collection = headers.get(CONTENT_ENCODING);
+		if (headers.containsKey(CONTENT_ENCODING) || headers.containsKey(CONTENT_ENCODING_LOWER)) {
```

Comment:
spring's `HttpHeaders` is the way to go for case insensitive headers, to avoid this kind of logic. It is used in `buildZuulRequestHeaders` above.